### ISSUE TITLE
Suggestion: state on the readme (or a separate file) the supported games / versions with MD5 hash

### ISSUE THREAD

**Issue by nighto** [hikari-no-yume/touchHLE#62]
Something that I see that emulation community doing, specially folks gathering around RetroAchievements, is to even though (obviously) not linking to pirated content etc., they provided hashes of the supported games (in this case, the games with Achievements) so you are positive that the version you have is actually supported. [This is how they do it.](https://docs.retroachievements.org/Game-Identification/)

So in a nutshell creating a file like supported-games.md:

```
SUPPORTED GAMES

The only currently supported games and versions are those listed down below, do not open bug reports for other games than those listed:

Super Monkey Ball 1.0 abcd1234
Super Monkey Ball 1.2 bcde2345
Super Monkey Ball 1.3 cdef3456
```

If you like the idea, I can create a pull request with such file.

**Comment by hikari-no-yume** [comment]
This is of course a good idea. Since I've gotten a second app working, I'm going to move the supported apps section into its own file anyway some time soon.

Unfortunately, to the best of my knowledge, you can't hash IPA files and get a consistent answer. There are many different hashes for the same version of an app, possibly an infinite number. I think this is because 1) due to the DRM, each IPA is personalised to the person who originally downloaded it, 2) decrypting the app might not fully depersonalise it, 3) IPA files are ZIP files and that format offers a lot of flexibility in how to encode data. I'm also can't just post the hashes of every decrypted version floating around online for reasons which are hopefully obvious and which I will not expand upon.

That being said, maybe someone else has come up with a reliable solution, or this project can solve it itself. touchHLE could hash the binary when it loads it, for example, which might avoid any personalisation problems.

**Comment by nighto** [comment]
Oh, interesting, I thought that after decryption it'd result into the same zip for everyone regardless of the account used to download. TIL.

Yeah in this case I think your idea would be great, having a `--hash` or whatever in which touchHLE would load the binary and calculate a MD5 hash instead of trying to run it.

**Comment by hikari-no-yume** [comment]
ZIP files are themselves part of the problem: they can contain the files in any order, the files can have all sorts of attributes associated with them, and there are an infinite number of possible compressions of the same file. With that said, maybe if you extract the files and are careful about which directories you scan, you could hash the individual files somehow?

**Comment by nighto** [comment]
Yeah, that could work out. So we could have a directory with metadata about the files inside the uncompressed ipas, calculating the sum for each file individually. I'll email you about joining the Discord so we can explore that idea.

**Comment by upintheairsheep** [comment]
> This is of course a good idea. Since I've gotten a second app working, I'm going to move the supported apps section into its own file anyway some time soon.
> 
> Unfortunately, to the best of my knowledge, you can't hash IPA files and get a consistent answer. There are many different hashes for the same version of an app, possibly an infinite number. I think this is because 1) due to the DRM, each IPA is personalised to the person who originally downloaded it, 2) decrypting the app might not fully depersonalise it, 3) IPA files are ZIP files and that format offers a lot of flexibility in how to encode data. I'm also can't just post the hashes of every decrypted version floating around online for reasons which are hopefully obvious and which I will not expand upon.
> 
> That being said, maybe someone else has come up with a reliable solution, or this project can solve it itself. touchHLE could hash the binary when it loads it, for example, which might avoid any personalisation problems.

Can you post footage of the game running on YouTube?

**Comment by hikari-no-yume** [comment]
touchHLE will now print the name and version of an app when running it (198846dc1ace4ad326a8e18cac40ad29e2704eae), so this should help reduce confusion somewhat. That also gives me a good place to put a hash of the binary, if I ever get round to it.
